### DISCUSSION TITLE
Time travel in DuckDB -- `SYSTEM TIME AS OF`

### DISCUSSION THREAD

**Discussion by david542542** [duckdb/duckdb#4000] [Ideas]
It would be very cool to be able to rollback to a previous state in the DuckDB database, if time travel is enabled. As an example of this from [BigQuery](https://cloud.google.com/bigquery/docs/time-travel):

```
SELECT *
FROM `mydataset.mytable`
  FOR SYSTEM_TIME AS OF TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 HOUR);
```

While DuckDB is not necessarily meant as a full-blown persistent store, I think it's a nice feature to have (when enabled) so that data can be rolled back to a particular place in time after the database was loaded (that is, it would obviously not persist after the connection was destroyed).

**Comment by Ture2019** [comment]
I have testet out the delta extension and added AS OF '2024-10-20 00:00:00' to the end of my query. But it didn´t work. I guess we have to wait for duckdb to come out with an working example for timetravel...
